components/clp-package-utils/clp_package_utils/general.py

```diff
@@ -574,3 +577,23 @@ def validate_path_for_container_mount(path: pathlib.Path) -> None:
                 f"Invalid path: `{path}` cannot be under '{prefix}' which may overlap with a path"
                 f" in the container."
             )
+
+
+def validate_dataset(clp_config: CLPConfig, dataset: typing.Optional[str]) -> None:
+    """
+    Checks if the provided dataset currently exists in the metadata database.
+    :param clp_config:
+    :param dataset:
+    """
+    if not dataset:
+        raise ValueError(f"`{dataset}` is not a valid dataset name.")
```

Comment:
It's weird that we let the dataset to be optional but immediately raise if it's None.

I kind of see that you are preparing for PR1023, but I feel that perhaps we can write code in the way such that caller with a dataset=None will never call this method.

That might turn into some code like the following, but it seems mypy is ok with it.
`
def validate_dataset(dataset: str):
    ...

def func_caller():
    ...
    input_arg: Optional[str]
    if input_arg is not None:
        validate_dataset(input_arg)
`